### PR TITLE
[risk=low][RW-7863] Fix contact email UX in AdminUserProfile

### PR DESCRIPTION
**Pull request by jmthibault79**
contactEmail

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/admin/admin-user-common.tsx [hunk 1]
```diff
@@ -356,7 +356,7 @@ export const updateAccountProperties = async (
     ]),
     contactEmail: getUpdatedProfileValue(oldProfile, updatedProfile, [
       'contactEmail',
-    ]),
+    ])?.trim(),
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-user-common.tsx:359]: not sure if needed, but it shouldn't hurt

File: ui/src/app/pages/admin/admin-user-profile.tsx [hunk 2]
```diff
@@ -468,11 +468,11 @@ export const AdminUserProfile = (spinnerProps: WithSpinnerOverlayProps) => {
   };
 
   const updateContactEmail = async (contactEmail: string) => {
+    updateProfile({ contactEmail });
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-user-profile.tsx:471]: This was the problem.  It waited for an API call to return before then updated state.  This caused delay and re-rendering.

Now it updates the email state immediately, for well behaved UX

File: ui/src/app/pages/admin/admin-user-profile.tsx [hunk 3]
```diff
@@ -587,7 +587,7 @@ export const AdminUserProfile = (spinnerProps: WithSpinnerOverlayProps) => {
               institutions={institutions}
               emailValidationStatus={emailValidationStatus}
               onChangeEmail={(contactEmail: string) =>
-                updateContactEmail(contactEmail.trim())
+                updateContactEmail(contactEmail)
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-user-profile.tsx:590]: aggressive onChange trimming can cause issues too

**Review by aweng98 — APPROVED** [review]
This fix is a life saver for the test! Thank you.

**Pull request merged**